### PR TITLE
AnimView tidying

### PR DESCRIPTION
**Pull request by TheCycoONE**
Mainly switches from separate layout arguments and ord bitflags to wxSizerFlags, which are clearer, particularly with respect to borders.

Also make some narrowing / sign conversions explicit.

### CODE REVIEW

File: AnimView/frmMain.cpp [hunk 1]
```diff
@@ -520,9 +531,10 @@ void frmMain::_onLastAnim(wxCommandEvent& evt) {
...
 
 void frmMain::_onAnimChar(wxCommandEvent& evt) {
-  long iAnim;
-  if (evt.GetString().ToLong(&iAnim)) {
-    if (iAnim >= 0 && iAnim < (long)m_oAnims.getAnimationCount()) {
+  long conv;
+  if (evt.GetString().ToLong(&conv)) {
+    size_t iAnim = static_cast<size_t>(conv);
+    if (iAnim >= 0 && iAnim < m_oAnims.getAnimationCount()) {
       _onAnimChange((size_t)iAnim);
```
Comment (Alberth289346) [inline AnimView/frmMain.cpp:538]: iAnim is already a `size_t`

That also makes the  `>= 0` check one line higher useless.

File: AnimView/frmMain.cpp [hunk 2]
```diff
@@ -66,25 +67,40 @@ BEGIN_EVENT_TABLE(frmMain, wxFrame)
...
   wxSizer* pMainSizer = new wxBoxSizer(wxHORIZONTAL);
 
   wxSizer* pSidebarSizer = new wxBoxSizer(wxVERTICAL);
 
-#define def wxDefaultPosition, wxDefaultSize
+  wxSizerFlags fillSizerFlags(0);
+  fillSizerFlags.Expand().Border(wxALL, 0);
+
+  wxSizerFlags fillBorderSizerFlags(0);
+  fillSizerFlags.Expand().Border(wxALL, 1);
```
Comment (Alberth289346) [inline AnimView/frmMain.cpp:86]: Missing "Border" in the variable name?

File: AnimView/frmMain.cpp [hunk 3]
```diff
@@ -95,232 +111,226 @@ frmMain::frmMain()
...
-  pSearchButtons->Add(new wxButton(this, ID_SEARCH_FRAME, L"Frame"), 0, wxALL,
-                      1);
-  pSearchButtons->Add(new wxButton(this, ID_SEARCH_SOUND, L"Sound"), 0, wxALL,
-                      1);
+  pSearchButtons->Add(new wxButton(this, ID_SEARCH_LAYER_ID, L"Layer/ID"),
+                      buttonSizerFlags);
+  pSearchButtons->Add(new wxButton(this, ID_SEARCH_FRAME, L"Frame"),
+                      buttonSizerFlags);
+  pSearchButtons->Add(new wxButton(this, ID_SEARCH_SOUND, L"Sound"),
+                      buttonSizerFlags);
```
Comment (Alberth289346) [inline AnimView/frmMain.cpp:322]: Sure these buttonSizer things are right, it's missing a `wxALIGN_CENTER_VERTICAL`

Comment (Alberth289346) [reply AnimView/frmMain.cpp:322]: ?

File: AnimView/frmMain.cpp (hunk 1 above)
Comment (TheCycoONE) [reply AnimView/frmMain.cpp:538]: Yep. Sloppy. I have to range check then cast

File: AnimView/frmMain.cpp (hunk 3 above)
Comment (TheCycoONE) [reply AnimView/frmMain.cpp:322]: Looks good to me. I thought it was better to be consistent with our buttons.

File: AnimView/frmMain.cpp (hunk 2 above)
Comment (TheCycoONE) [reply AnimView/frmMain.cpp:86]: Alright. I bet that changed appearances. Didn't notice from a sight inspection.

File: AnimView/frmMain.cpp (hunk 3 above)
Comment (Alberth289346) [reply AnimView/frmMain.cpp:322]: Likely a good idea, I just verified whether the flags were the same before and after the change, and these failed that test :)

File: AnimView/frmMain.cpp [hunk 4]
```diff
@@ -520,11 +531,13 @@ void frmMain::_onLastAnim(wxCommandEvent& evt) {
...
-  if (evt.GetString().ToLong(&iAnim)) {
-    if (iAnim >= 0 && iAnim < (long)m_oAnims.getAnimationCount()) {
-      _onAnimChange((size_t)iAnim);
+  long conv;
+  if (evt.GetString().ToLong(&conv)) {
+    if (conv < 0 || conv >= m_oAnims.getAnimationCount()) {
+      return;
     }
+    size_t iAnim = static_cast<size_t>(conv);
+    _onAnimChange((size_t)iAnim);
```
Comment (Alberth289346) [inline AnimView/frmMain.cpp:540]: Still doing a cast to its own type.

**Comment by Alberth289346** [comment]
Looks fine to me

**Pull request merged**